nodes/basic-pow/src/chain_spec.rs

```diff
@@ -0,0 +1,107 @@
+use sp_core::{Pair, Public, sr25519};
+use sc_service;
+use sp_runtime::traits::{Verify, IdentifyAccount};
+use runtime::{AccountId, GenesisConfig, Signature};
+use runtime_genesis::testnet_genesis;
+
+// Note this is the URL for the telemetry server
+//const STAGING_TELEMETRY_URL: &str = "wss://telemetry.polkadot.io/submit/";
+
+/// Specialized `ChainSpec`. This is a specialization of the general Substrate `ChainSpec` type.
+pub type ChainSpec = sc_service::ChainSpec<GenesisConfig>;
+
+/// The chain specification option. This is expected to come in from the CLI and
+/// is little more than one of a number of alternatives which can easily be converted
+/// from a string (`--chain=...`) into a `ChainSpec`.
+#[derive(Clone, Debug)]
+pub enum Alternative {
+	/// Whatever the current runtime is, with just Alice as an auth.
+	Development,
+	/// Whatever the current runtime is, with simple Alice/Bob auths.
+	LocalTestnet,
+}
+
+/// Helper function to generate a crypto pair from seed
+pub fn get_from_seed<TPublic: Public>(seed: &str) -> <TPublic::Pair as Pair>::Public {
+	TPublic::Pair::from_string(&format!("//{}", seed), None)
+		.expect("static values are valid; qed")
+		.public()
+}
```

Comment:
The trait `Pair` has a `Public` type, so this is kind of roundabout. 
I've been working on crypto traits here [#4954](https://github.com/paritytech/substrate/pull/4954).

```suggestion
pub fn get_from_seed<TPair: Pair>(seed: &str) -> TPair::Public {
	TPair::from_string(&format!("//{}", seed), None)
		.expect("static values are valid; qed")
		.public()
}
```

Reply:
I've made this change here in the PoW node, and it works as expected. Because I took this original code straight from the node template, I tried to make the same changes to the Substrate repo here https://github.com/paritytech/substrate/blob/master/bin/node-template/node/src/chain_spec.rs#L28-L50 and it didn't work. I realize my node is simpler because it doesn't have aura or grandpa keys. Is this change only possible in this simplified situation?

Reply:
yeah so these lines are the problem

https://github.com/paritytech/substrate/blob/f546ddb4c4760b0d7b17fea5cfe72adb69d02139/bin/node-template/node/src/chain_spec.rs#L46-L49

`AuraId` and `GrandpaId` is the publick key type of the pair and so the above code isn't really roundabout but intentional.